src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1855,11 +1822,19 @@ private ICaseClauseOperation CreateBoundPatternSwitchLabelOperation(BoundPattern
             Optional<object> constantValue = default(Optional<object>);
             bool isImplicit = boundPatternSwitchLabel.WasCompilerGenerated;
 
-            if (boundPatternSwitchLabel.Pattern.Kind == BoundKind.DiscardPattern)
+            if (boundPatternSwitchLabel.Syntax.Kind() == SyntaxKind.DefaultSwitchLabel)
             {
-                // Default switch label in pattern switch statement is represented as a default case clause.
+                Debug.Assert(boundPatternSwitchLabel.Pattern.Kind == BoundKind.DiscardPattern);
                 return new DefaultCaseClause(_semanticModel, syntax, type, constantValue, isImplicit);
             }
+            else if (boundPatternSwitchLabel.Guard == null &&
+                     boundPatternSwitchLabel.Pattern.Kind == BoundKind.ConstantPattern &&
```

Comment:
>boundPatternSwitchLabel.Pattern.Kind == BoundKind.ConstantPattern && [](start = 21, length = 68)

Can be removed given the following `is` pattern. Or alternatively, `is` pattern can be replaced with explicit cast.

Reply:
The Kind test is faster and avoids the need for the type test.